resources/assets/components/CampaignSingle/index.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { flatMap, keyBy, map, sample, forEach, reject, filter } from 'lodash';
+
+import InboxItem from '../InboxItem';
+import ModalContainer from '../ModalContainer';
+import HistoryModal from '../HistoryModal';
+import PostFilter from '../PostFilter';
+import StatusCounter from '../StatusCounter';
+import { RestApiClient} from '@dosomething/gateway';
+
+class CampaignSingle extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const posts = keyBy(flatMap(props.signups, signup => {
+      return signup.posts.map(post => {
+        post.signup = signup;
+        return post;
+      });
+    }), 'id');
+
+    this.state = {
+      posts: posts,
+      filteredPosts: posts,
+      postTotals: props.post_totals,
+      displayHistoryModal: false,
+      historyModalId: null,
```

Comment:
Could it make sense to rename this to `showHistoryForPostId` or something like that? At first glance I thought this was referring to a HTML `id` attribute.